src/Constants/authorization.js

```diff
@@ -0,0 +1,117 @@
+/* 
+To Dos:
+1. Testing
+*/
+
+// Spotify application details
+const stateKey = 'spotify_auth_state';
+const CLIENT_ID = 'c73c8dd43ae64b7c82a1e3b355cda443';
+// const REDIRECT_URI = 'http://localhost:8080';
+const REDIRECT_URI = 'https://curious-salamander-cadf27.netlify.app/';
+const scope = 'playlist-modify-private playlist-modify-public user-read-private user-read-email';
+
+// Function to generate a random string
+function generateRandomString(length) {
+    let text = '';
+    let possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+
+    for (let i = 0; i < length; i++) {
+        text += possible.charAt(Math.floor(Math.random() * possible.length));
+    }
+    return text;
+};
+
+// Function to set the state in localStorage
+function setStateInStorage(stateKey) {
+    const state = generateRandomString(16);
+    localStorage.setItem(stateKey, state);
+};
+
+// Function to handle login
+const handleLogin = (e) => {    
+    e.preventDefault();
+
+    //Set the state in localStorage
+    setStateInStorage(stateKey);
+
+    // Get the generated state from localStorage
+    const storedState = localStorage.getItem(stateKey);
+
+    // Build the authorization URL using the generated state
+    let url = 'https://accounts.spotify.com/authorize';
+    url += '?client_id=' + encodeURIComponent(CLIENT_ID);
+    url += '&response_type=token';
+    url += '&redirect_uri=' + encodeURIComponent(REDIRECT_URI);
+    url += '&state=' + encodeURIComponent(storedState);
+    url += '&scope=' + encodeURIComponent(scope);
+
+    // Redirect the user to the Spotify auhorization page
+    window.location.href = url;
+};
+
+// Implicit Grant Flow authorization
+const authorization = () => {
```

Comment:
nitpick: authorization as a function name doesn't make it clear to me what the function does. Maybe something like `retrieveTokenFromUrlHash`?

Reply:
Done